starter_code/src/movies.js

```diff
@@ -1,20 +1,190 @@
+function ratesAverage(movies) {
+    var sum = movies.reduce(function (acc, movie) {
+        // return acc + Number(movie.rate);
+        return acc + parseFloat(movie.rate);
+    }, 0);
+    // console.log((sum / movies.length).toFixed(2));
+    return (Number((sum / movies.length).toFixed(2)));
+}
+ratesAverage(movies);
 
 
 // Get the average of Drama Movies
+/*
+    function dramaMoviesRate(movies) {
+        var filterMoviesDrama = movies.filter(function (item) {
+            return item.genre.some(
+                function (item) {
+                    return item === 'Drama';
+                })
+        });
+        return filterMoviesDrama.length ? ratesAverage(filterMoviesDrama) : undefined;
+    }
+*/
+
+function dramaMoviesRate(movies) {
+    var arrayDramaMovies = movies.filter(function (movie) {
+        if (movie.genre.indexOf("Drama") != -1) {
+            return movie;
+        }
+        else {
+            return undefined;
+        }
+    });
+    // Devolvemos undefined si no hay películas de drama
+    if (arrayDramaMovies.length === 0) {
+        return undefined;
+    }
+    else {
+        var sumDramaMovies2 = arrayDramaMovies.reduce(function (acc, movie) {
+            return acc + Number(movie.rate);
+        }, 0);
+
+        return Number((sumDramaMovies2 / arrayDramaMovies.length).toFixed(2));
+    }
+
+}
 
+dramaMoviesRate(movies);
 
 // Order by time duration, in growing order
 
+/*
+    function orderByDuration(movies) {
+        var moviesParseToMinutes = turnHoursToMinutes(movies);
+
+        return moviesParseToMinutes.sort(function (item1, item2) {
+            return item2.duration - item1.duration;
+        });
+    }
+*/
+
+function orderByDuration(movies) {
+    var newOrderMovies = movies.sort(function (movieRate1, movieRate2) {
+        if (movieRate1.duration === movieRate2.duration ) {
+            return movieRate1.title < movieRate2.title ? -1 : 1;
+        }
+        else {
+            return movieRate1.duration - movieRate2.duration;
+        }
+    });
+    return newOrderMovies;
+}
+orderByDuration(turnHoursToMinutes(movies));
+
 
 // How many movies did STEVEN SPIELBERG
 
+function howManyMovies(movieArray) {
+    if (!movieArray.length) {
+        return undefined;
+    }
+
+    var steveMovies = movieArray.filter(function (item) {
+        return item.director === 'Steven Spielberg' &&
+            item.genre.some(
+                function (item) {
+                    return item === 'Drama';
+                });
+    });
+
+    return 'Steven Spielberg directed ' + steveMovies.length + ' drama movies!';
+}
 
 // Order by title and print the first 20 titles
 
+function orderAlphabetically(moviesArray) {
+    var moviesTitle = moviesArray.map(function (item) {
+        return item.title;
+    }).sort(function (title1, title2) {
+        return title1.localeCompare(title2)
+    });
+
+    return moviesTitle.length > 20 ? moviesTitle.filter((item, i) => i < 20) : moviesTitle;
```

Comment:
No me vale, cambiad la línea 146, No uséis arrow functions ni metáis el filter en el ternario para reducir a una línea por ahora